Source/Tabs/DataProviders/ShoppingList.lua

```diff
@@ -0,0 +1,129 @@
+local SHOPPING_LIST_TABLE_LAYOUT = {
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerParameters = { "minPrice" },
+    headerText = "Price",
+    cellTemplate = "AuctionatorPriceCellTemplate",
+    cellParameters = { "minPrice" },
+    width = 140
+  },
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerParameters = { "name" },
+    headerText = "Name",
+    cellTemplate = "AuctionatorItemKeyCellTemplate"
+  },
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerText = "Available",
+    headerParameters = { "totalQuantity" },
+    cellTemplate = "AuctionatorStringCellTemplate",
+    cellParameters = { "totalQuantity" },
+    width = 70
+  }
+}
+
+local ITEM_KEY_EVENT = "ITEM_KEY_ITEM_INFO_RECEIVED"
+
+ShoppingListDataProviderMixin = CreateFromMixins(DataProviderMixin)
+
+function ShoppingListDataProviderMixin:OnLoad()
+  Auctionator.Debug.Message("ShoppingListDataProviderMixin:OnLoad()")
+
+  self.pendingItemIds = {}
+
+  DataProviderMixin.OnLoad(self)
+
+  self:GetParent():Register( self, {
+    Auctionator.ShoppingLists.Events.ListSearchStarted,
+    Auctionator.ShoppingLists.Events.ListSearchEnded,
+    Auctionator.ShoppingLists.Events.ListSearchIncrementalUpdate
+  })
+end
+
+function ShoppingListDataProviderMixin:OnEvent(event, ...)
+  if event == ITEM_KEY_EVENT then
+    local itemId = ...
+
+    for _, entry in ipairs(self.results) do
+      if entry.itemKey.itemID == itemId then
+        self:FetchItemKey(entry)
+        return
```

Comment:
Multiple itemKeys can have the same itemID. Don't return after the first one found:
![image](https://user-images.githubusercontent.com/60280559/78905358-c4d0c500-7a75-11ea-8b29-c1c11f68dfb4.png)

Reply:
Removed `return` (will push after addressing other comments)